// @ionic-native/sqlite-db-copy/index.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Injectable } from '@angular/core';
import { Cordova, IonicNativePlugin, Plugin } from '@ionic-native/core';
/**
 * @name Sqlite Db Copy
 * @description
 * This plugin does something
 *
 * @usage
 * ```typescript
 * import { SqliteDbCopy } from '@ionic-native/sqlite-db-copy';
 *
 *
 * constructor(private sqliteDbCopy: SqliteDbCopy) { }
 *
 * ...
 *
 *
 * this.sqliteDbCopy.copy('sample.db', 0)
 *   .then((res: any) => console.log(res))
 *   .catch((error: any) => console.error(error));
 *
 * ```
 */
var SqliteDbCopy = (function (_super) {
    __extends(SqliteDbCopy, _super);
    function SqliteDbCopy() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
     * Copy database from www directory to device SQLite DB location
     * (for ios only, use 0 for Android)
     * location = 0; // (Disable iCloud Backup) If set will copy the database to Library/LocalDatabase. The database will not be synced by the iCloud Backup.
     * or
     * location = 1; // If set will copy the database to Library folder instead of Documents folder.
     * or
     * location = 2; // It will copy the database in the default SQLite Database directory. This is the default location for database
     * @param dbname {string} Database file name available in www diretory with extension.The plugin will look for and copy the file according to the filename provided here. And the same file name should be used while opening the database via SQLitePlugin
     * @param location {number} Location where to copy the database for app (only applicable for ios).For Android pass 0.
     * @return {Promise<any>} Returns a promise that resolves when something happens
     */
    /**
       * Copy database from www directory to device SQLite DB location
       * (for ios only, use 0 for Android)
       * location = 0; // (Disable iCloud Backup) If set will copy the database to Library/LocalDatabase. The database will not be synced by the iCloud Backup.
       * or
       * location = 1; // If set will copy the database to Library folder instead of Documents folder.
       * or
       * location = 2; // It will copy the database in the default SQLite Database directory. This is the default location for database
       * @param dbname {string} Database file name available in www diretory with extension.The plugin will look for and copy the file according to the filename provided here. And the same file name should be used while opening the database via SQLitePlugin
       * @param location {number} Location where to copy the database for app (only applicable for ios).For Android pass 0.
       * @return {Promise<any>} Returns a promise that resolves when something happens
       */
    SqliteDbCopy.prototype.copy = /**
       * Copy database from www directory to device SQLite DB location
       * (for ios only, use 0 for Android)
       * location = 0; // (Disable iCloud Backup) If set will copy the database to Library/LocalDatabase. The database will not be synced by the iCloud Backup.
       * or
       * location = 1; // If set will copy the database to Library folder instead of Documents folder.
       * or
       * location = 2; // It will copy the database in the default SQLite Database directory. This is the default location for database
       * @param dbname {string} Database file name available in www diretory with extension.The plugin will look for and copy the file according to the filename provided here. And the same file name should be used while opening the database via SQLitePlugin
       * @param location {number} Location where to copy the database for app (only applicable for ios).For Android pass 0.
       * @return {Promise<any>} Returns a promise that resolves when something happens
       */
    function (dbname, location) {
        return;
    };
    /**
     * Check if database is available at external storage. Useful when trying to copying the database from device sdcard.
     * @param dbname {string} Name of the database file which is available on external or intenral storage
     * @param source {string} Full native path for the folder in which db file is present. The "/" must be added at the end of path. For.eg. /sdcard/mydb/. Should not include dbname.
     * @return {Promise<any>}
     */
    /**
       * Check if database is available at external storage. Useful when trying to copying the database from device sdcard.
       * @param dbname {string} Name of the database file which is available on external or intenral storage
       * @param source {string} Full native path for the folder in which db file is present. The "/" must be added at the end of path. For.eg. /sdcard/mydb/. Should not include dbname.
       * @return {Promise<any>}
       */
    SqliteDbCopy.prototype.checkDbOnStorage = /**
       * Check if database is available at external storage. Useful when trying to copying the database from device sdcard.
       * @param dbname {string} Name of the database file which is available on external or intenral storage
       * @param source {string} Full native path for the folder in which db file is present. The "/" must be added at the end of path. For.eg. /sdcard/mydb/. Should not include dbname.
       * @return {Promise<any>}
       */
    function (dbname, source) {
        return;
    };
    /**
     * Copy database which is available on the device external/internal storage to app default db location
     * (for ios only, use 0 for Android)
     * location = 0; // (Disable iCloud Backup) If set will copy the database to Library/LocalDatabase. The database will not be synced by the iCloud Backup.
     * or
     * location = 1; // If set will copy the database to Library folder instead of Documents folder.
     * or
     * location = 2; // It will copy the database in the default SQLite Database directory. This is the default location for database
     * @param dbname {string} Is the name of the database you want to copy. The plugin will look for and copy the file according to the filename provided here. And the same file name should be used while opening the database via SQLitePlugin
     * @param location {number} Location where to copy the database for app (only applicable for ios). For Android pass 0.
     * @param source {string} Source File location like /sdcard/mydb/db.db. Please provide a valid existing location and the dbname should be present in the path.
     * @param deleteolddb {boolean} A boolean value if set to true, will delete the existing db from the local app database folder before copying the new db. Please provide proper boolean value true or false;
     * @return {Promise<any>}
     */
    /**
       * Copy database which is available on the device external/internal storage to app default db location
       * (for ios only, use 0 for Android)
       * location = 0; // (Disable iCloud Backup) If set will copy the database to Library/LocalDatabase. The database will not be synced by the iCloud Backup.
       * or
       * location = 1; // If set will copy the database to Library folder instead of Documents folder.
       * or
       * location = 2; // It will copy the database in the default SQLite Database directory. This is the default location for database
       * @param dbname {string} Is the name of the database you want to copy. The plugin will look for and copy the file according to the filename provided here. And the same file name should be used while opening the database via SQLitePlugin
       * @param location {number} Location where to copy the database for app (only applicable for ios). For Android pass 0.
       * @param source {string} Source File location like /sdcard/mydb/db.db. Please provide a valid existing location and the dbname should be present in the path.
       * @param deleteolddb {boolean} A boolean value if set to true, will delete the existing db from the local app database folder before copying the new db. Please provide proper boolean value true or false;
       * @return {Promise<any>}
       */
    SqliteDbCopy.prototype.copyDbFromStorage = /**
       * Copy database which is available on the device external/internal storage to app default db location
       * (for ios only, use 0 for Android)
       * location = 0; // (Disable iCloud Backup) If set will copy the database to Library/LocalDatabase. The database will not be synced by the iCloud Backup.
       * or
       * location = 1; // If set will copy the database to Library folder instead of Documents folder.
       * or
       * location = 2; // It will copy the database in the default SQLite Database directory. This is the default location for database
       * @param dbname {string} Is the name of the database you want to copy. The plugin will look for and copy the file according to the filename provided here. And the same file name should be used while opening the database via SQLitePlugin
       * @param location {number} Location where to copy the database for app (only applicable for ios). For Android pass 0.
       * @param source {string} Source File location like /sdcard/mydb/db.db. Please provide a valid existing location and the dbname should be present in the path.
       * @param deleteolddb {boolean} A boolean value if set to true, will delete the existing db from the local app database folder before copying the new db. Please provide proper boolean value true or false;
       * @return {Promise<any>}
       */
    function (dbname, location, source, deleteolddb) {
        return;
    };
    /**
     * Copy the app database to external/internal storage on the device.
     * (for ios only, use 0 for Android)
     * location = 0; //It will copy the database from Library/LocalDatabase location.
     * or
     * location = 1; // It will copy the database from Library folder instead.
     * or
     * location = 2; // It will copy the database from the default SQLite Database directory. This is the default location for database
     * @param dbname {string} Is the name of the database you want to copy. The plugin will look for and copy the file according to the filename provided here. And the same file name should be used while opening the database via SQLitePlugin.
     * @param location {number} Location where to copy the database for app (only applicable for ios). For Android pass 0.
     * @param destination {string} Destination File location like /sdcard/mydb/ Please provide a valid existing location and "/" should be present at the end of the path. Do not append db name in the path.
     * @param overwrite {boolean}  If set to true, then will replace the file at the destination. Otherwise will throw an error, if destination file already exists.
     * @return {Promise<any>}
     */
    /**
       * Copy the app database to external/internal storage on the device.
       * (for ios only, use 0 for Android)
       * location = 0; //It will copy the database from Library/LocalDatabase location.
       * or
       * location = 1; // It will copy the database from Library folder instead.
       * or
       * location = 2; // It will copy the database from the default SQLite Database directory. This is the default location for database
       * @param dbname {string} Is the name of the database you want to copy. The plugin will look for and copy the file according to the filename provided here. And the same file name should be used while opening the database via SQLitePlugin.
       * @param location {number} Location where to copy the database for app (only applicable for ios). For Android pass 0.
       * @param destination {string} Destination File location like /sdcard/mydb/ Please provide a valid existing location and "/" should be present at the end of the path. Do not append db name in the path.
       * @param overwrite {boolean}  If set to true, then will replace the file at the destination. Otherwise will throw an error, if destination file already exists.
       * @return {Promise<any>}
       */
    SqliteDbCopy.prototype.copyDbToStorage = /**
       * Copy the app database to external/internal storage on the device.
       * (for ios only, use 0 for Android)
       * location = 0; //It will copy the database from Library/LocalDatabase location.
       * or
       * location = 1; // It will copy the database from Library folder instead.
       * or
       * location = 2; // It will copy the database from the default SQLite Database directory. This is the default location for database
       * @param dbname {string} Is the name of the database you want to copy. The plugin will look for and copy the file according to the filename provided here. And the same file name should be used while opening the database via SQLitePlugin.
       * @param location {number} Location where to copy the database for app (only applicable for ios). For Android pass 0.
       * @param destination {string} Destination File location like /sdcard/mydb/ Please provide a valid existing location and "/" should be present at the end of the path. Do not append db name in the path.
       * @param overwrite {boolean}  If set to true, then will replace the file at the destination. Otherwise will throw an error, if destination file already exists.
       * @return {Promise<any>}
       */
    function (dbname, location, destination, overwrite) {
        return;
    };
    /**
     * This method allows you to remove the database from the apps default database storage location.
     * (for ios only, use 0 for Android)
     * location = 0; // (Disable iCloud Backup) If set will copy the database to Library/LocalDatabase. The database will not be synced by the iCloud Backup.
     * or
     * location = 1; // If set will copy the database to Library folder instead of Documents folder.
     * or
     * location = 2; // It will copy the database in the default SQLite Database directory. This is the default location for database
     * @param dbname {string} Is the name of the database you want to remove. If the database file is having any extension, please provide that also.
     * @param location {number} Location where to copy the database for app (only applicable for ios).For Android pass 0.
     * @return {Promise<any>}
     */
    /**
       * This method allows you to remove the database from the apps default database storage location.
       * (for ios only, use 0 for Android)
       * location = 0; // (Disable iCloud Backup) If set will copy the database to Library/LocalDatabase. The database will not be synced by the iCloud Backup.
       * or
       * location = 1; // If set will copy the database to Library folder instead of Documents folder.
       * or
       * location = 2; // It will copy the database in the default SQLite Database directory. This is the default location for database
       * @param dbname {string} Is the name of the database you want to remove. If the database file is having any extension, please provide that also.
       * @param location {number} Location where to copy the database for app (only applicable for ios).For Android pass 0.
       * @return {Promise<any>}
       */
    SqliteDbCopy.prototype.remove = /**
       * This method allows you to remove the database from the apps default database storage location.
       * (for ios only, use 0 for Android)
       * location = 0; // (Disable iCloud Backup) If set will copy the database to Library/LocalDatabase. The database will not be synced by the iCloud Backup.
       * or
       * location = 1; // If set will copy the database to Library folder instead of Documents folder.
       * or
       * location = 2; // It will copy the database in the default SQLite Database directory. This is the default location for database
       * @param dbname {string} Is the name of the database you want to remove. If the database file is having any extension, please provide that also.
       * @param location {number} Location where to copy the database for app (only applicable for ios).For Android pass 0.
       * @return {Promise<any>}
       */
    function (dbname, location) {
        return;
    };
    SqliteDbCopy.decorators = [
        { type: Injectable },
    ];
    __decorate([
        Cordova({
            successIndex: 2,
            errorIndex: 3
        }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [String, Number]),
        __metadata("design:returntype", Promise)
    ], SqliteDbCopy.prototype, "copy", null);
    __decorate([
        Cordova({
            successIndex: 2,
            errorIndex: 3
        }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [String, String]),
        __metadata("design:returntype", Promise)
    ], SqliteDbCopy.prototype, "checkDbOnStorage", null);
    __decorate([
        Cordova({
            successIndex: 4,
            errorIndex: 5
        }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [String, Number, String, Boolean]),
        __metadata("design:returntype", Promise)
    ], SqliteDbCopy.prototype, "copyDbFromStorage", null);
    __decorate([
        Cordova({
            successIndex: 4,
            errorIndex: 5
        }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [String, Number, String, Boolean]),
        __metadata("design:returntype", Promise)
    ], SqliteDbCopy.prototype, "copyDbToStorage", null);
    __decorate([
        Cordova({
            successIndex: 2,
            errorIndex: 3
        }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [String, Number]),
        __metadata("design:returntype", Promise)
    ], SqliteDbCopy.prototype, "remove", null);
    /**
     * @name Sqlite Db Copy
     * @description
     * This plugin does something
     *
     * @usage
     * ```typescript
     * import { SqliteDbCopy } from '@ionic-native/sqlite-db-copy';
     *
     *
     * constructor(private sqliteDbCopy: SqliteDbCopy) { }
     *
     * ...
     *
     *
     * this.sqliteDbCopy.copy('sample.db', 0)
     *   .then((res: any) => console.log(res))
     *   .catch((error: any) => console.error(error));
     *
     * ```
     */
    SqliteDbCopy = __decorate([
        Plugin({
            pluginName: 'cordova-plugin-dbcopy',
            plugin: 'cordova-plugin-dbcopy',
            pluginRef: 'window.plugins.sqlDB',
            repo: 'https://github.com/an-rahulpandey/cordova-plugin-dbcopy',
            platforms: ['Android', 'iOS']
        })
    ], SqliteDbCopy);
    return SqliteDbCopy;
}(IonicNativePlugin));
export { SqliteDbCopy };
